src/aave-v3/SupplyVaultBase.sol

```diff
@@ -99,7 +95,7 @@ abstract contract SupplyVaultBase is ERC4626UpgradeableSafe, OwnableUpgradeable
         );
 
         return
-            supplyBalance.onPool.rayMul(pool.getReserveNormalizedIncome(asset())) +
+            supplyBalance.onPool.rayMul(morpho.poolIndexes(poolTokenMem).poolSupplyIndex) +
             supplyBalance.inP2P.rayMul(morpho.p2pSupplyIndex(poolTokenMem));
```

Comment:
This is not related to this PR, but I think that there might be an issue here:

When depositing, this is called by `convertToShares` before updating the indexes on Morpho, meaning that they are potentially smaller than what they should. This means that if you instantly withdraw after, because the indexes would have been updated but the deposit, you will withdraw more than what you should be able to withdraw, stealing profits from other users. I'm not able to say if the amount that you steal is consequent or not, as it depends on a lot of things..

Reply:
Yes but it was the same before as we were using p2pSupplyIndex no?

And the function must stay public I guess..

Reply:
Shoud we add a call to update indexes before?

Reply:
Yes it was the same, this issue is not related to this PR but is still an issue. If you want to be quick on this one, we can move into a Github issue but this should be discussed.

Reply:
I think we should

Reply:
Calling `updateIndexes` fixes the issue yes. For not that much added gas cost, because it warms the storage slots.

Reply:
Let's do that!

Reply:
This means that this function will no longer be a `view`

Reply:
ah yes...

Reply:
not a solution then...

Reply:
We could update the indexes in the `_deposit` and `_withdraw` functions at the start instead. The function `totalAssets` would be a bit wrong if used from the front-end, but it will still be a view function.
(btw nice catch @MathisGD !)

Reply:
Not sure it's possible @QGarchery given how these functions are written.

Reply:
ok found the solution it's trivial

Reply:
Ah you are right, we may have to override other functions from ERC4626Upgradeable

Reply:
But you're right @QGarchery sorry I misunderstood your point. We can do it in our functions!

Reply:
I think we can do the following: 
- override the public non-view functions such as `mint`
- add an update of the indexes at the start
- inline what we wrote when overriding `_deposit` and `_withdraw` in those functions to make it cleaner

Reply:
Hmm i think there's is something simple, let me push something.

Reply:
I don't think that it works like this